crowbar_framework/app/models/service_object.rb

```diff
@@ -1173,6 +1173,10 @@ def apply_role(role, inst, in_queue)
 
       nodes_in_batch = []
 
+      # get databag to remember potential removal of a role
+      databag = Chef::DataBag.load "barclamps/#{@bc_name}"
+      databag["crowbar"]["roles_to_remove"] = []
```

Comment:
@vuntz good catch. i guess i can reach that `elements` over the `ProposalObject`.
but this will be dropped soon IIUC.
shall I go the way over the `ProposalObject` for now?